mm2src/coins/utxo.rs

```diff
@@ -2228,3 +782,104 @@ fn kmd_interest(height: Option<u64>, value: u64, lock_time: u64, current_time: u
     minutes -= 59;
     (value / 10512000) * minutes
 }
+
+/// Denominate BigDecimal amount of coin units to satoshis
+pub fn sat_from_big_decimal(amount: &BigDecimal, decimals: u8) -> Result<u64, String> {
+    (amount * BigDecimal::from(10u64.pow(decimals as u32))).to_u64().ok_or(ERRL!("Could not get sat from amount {} with decimals {}", amount, decimals))
+}
+
+pub(crate) fn sign_tx(
+    unsigned: TransactionInputSigner,
+    key_pair: &KeyPair,
+    prev_script: Script,
+    signature_version: SignatureVersion,
+    fork_id: u32,
+) -> Result<UtxoTx, String> {
+    let mut signed_inputs = vec![];
+    for (i, _) in unsigned.inputs.iter().enumerate() {
+        signed_inputs.push(
+            try_s!(p2pkh_spend(&unsigned, i, key_pair, &prev_script, signature_version, fork_id))
+        );
+    }
+    Ok(UtxoTx {
+        inputs: signed_inputs,
+        n_time: unsigned.n_time,
+        outputs: unsigned.outputs.clone(),
+        version: unsigned.version,
+        overwintered: unsigned.overwintered,
+        lock_time: unsigned.lock_time,
+        expiry_height: unsigned.expiry_height,
+        join_splits: vec![],
+        shielded_spends: vec![],
+        shielded_outputs: vec![],
+        value_balance: 0,
+        version_group_id: unsigned.version_group_id,
+        binding_sig: H512::default(),
+        join_split_sig: H512::default(),
+        join_split_pubkey: H256::default(),
+        zcash: unsigned.zcash,
+        str_d_zeel: unsigned.str_d_zeel,
+    })
+}
+
+async fn send_outputs_from_my_address_impl<T>(coin: T, outputs: Vec<TransactionOutput>)
+                                              -> Result<UtxoTx, String>
+    where T: UtxoArcGetter + UtxoArcCommonOps {
+    let _utxo_lock = UTXO_LOCK.lock().await;
+    // TODO replace list_unspent_ordered with ordered_mature_unspents
+    let unspents = try_s!(coin.arc().rpc_client.list_unspent_ordered(&coin.arc().my_address).map_err(|e| ERRL!("{}", e)).compat().await);
+    // TODO check for QRC20
```

Comment:
TODO?

Reply:
The generate_transaction method takes `gas_fee:  Optional<u64>`. We can use default gas limit and gas price for the swaps like [here](https://github.com/KomodoPlatform/atomicDEX-API/blob/qrc20/mm2src/coins/utxo/qrc20.rs#L458).